zcash_primitives/src/transaction/mod.rs

```diff
@@ -67,7 +68,7 @@ const ZFUTURE_TX_VERSION: u32 = 0x0000FFFF;
 ///   that have been mined.
 /// - For v5 transactions onwards, this identifier is derived only from "effecting" data,
 ///   and is non-malleable in all contexts.
-#[derive(Clone, Copy, PartialOrd, Ord, PartialEq, Eq, Hash)]
+#[derive(Clone, Copy, PartialOrd, Ord, PartialEq, Eq, Hash, Serialize, Deserialize)]
```

Comment:
This should use hand-written `Serialize` and `Deserialize` impls that serialize to the canonical txid text string representation.

Reply:
Why? Also I dont see the definition for the canonical string rep - could you point me to it? I do see `Display` implemented so maybe that is it? Is there a `FromStr` for `TxId` anywhere?

Reply:
We haven't needed to implement parsing, but yes the string representation used in the `Display` impl is canonical.